abstractions/dotnet/src/QueryParametersBase.cs

```diff
@@ -23,7 +24,7 @@ public void AddQueryParameters(IDictionary<string, object> target)
                                         .GetProperties()
                                         .Where(x => !target.ContainsKey(x.Name)))
             {
-                target.Add(property.Name, property.GetValue(this));
+                target.Add(property.Name.ToFirstCharacterLowerCase(), property.GetValue(this));
```

Comment:
The reason for this is that I think it's safer/consistent to default changing the name to the lowerCase of the first letter for now if the queryParameter is inserted through this "reflection of property names" route. 
We should probably investigate the generating of Attributes so that we can know when to add the '$' character.

This can still be overridden if they manipulate the `RequestInformation` instance directly though the `QueryParameters` dictionary without using the RequestBuilders and since the dictionary is set to [IgnoreCase](https://github.com/microsoft/kiota/blob/f258da5743504b4b355f90803ff571b6599e5cad/abstractions/dotnet/src/RequestInformation.cs#L54), they should still be able to access it even if this change goes through.

Reply:
the $ sign in front of OData query parameters is not required anymore since v4. I don't think that's a concern we need to have anymore. On top of that, Kiota is not supposed to know about any OData convention.

Reply:
Thanks for the tip!